### PR TITLE
support for user's preferred locale option

### PR DESCRIPTION
**Pull request by i-just**
### Description
Added support for saving user’s preferred locale.

### Related issues
#612

### CODE REVIEW

File: src/services/Process.php [hunk 1]
```diff
@@ -473,6 +473,15 @@ public function processFeed($step, $feed, &$processedElementIds, $feedData = nul
 
         // Save the element
         if ($this->_service->save($element, $feed)) {
+
+            // save user's preferences only after user has been successfully saved
+            if (isset($attributeData['preferredLocale'])) {
```
Comment (angrybrad) [inline src/services/Process.php:478]: Should probably also check that it’s an actual user element here in the case where there’s a custom field named "preferredLocale" on any other element?

Comment (i-just) [reply src/services/Process.php]: I was going to say that this is only kicking in for attribute data, not custom fields data. Custom fields data is in the `$fieldData` array. And User element is the only one with the `preferredLocale` attribute, but actually, you could create your own custom element with that attribute, so yes, I added that check. Thanks!

**Review by angrybrad — COMMENTED** [review]
Probably missing something, but where does `parsePreferredLocale()` actually get called from? I don’t see it in any base classes and nothing in Feed Me’s User element calls it.

**Comment by i-just** [comment]
`services/Process->processFeed()` calls `base\Element->parseAttribute()` which calls a method called `'parse' . ucwords($fieldHandle)`. That’s how most attributes are dealt with.

https://github.com/craftcms/feed-me/blob/main/src/services/Process.php#L364
https://github.com/craftcms/feed-me/blob/main/src/base/Element.php#L100-L107

Screenshot 2023-04-28 at 08 02 47

**Comment by angrybrad** [comment]
@i-just thank you!

**Comment by gschratzer** [comment]
just i was confused about the fieldname PreferredLocale, that is the functionname but in my csv it is working fine in my moodle [4.1.6 (Build: 20231009)] perfect if i have a field lang with the isocode
en, fr, ro, ru,  etc

**Pull request merged**